site/pages/token-revokes.js

```diff
@@ -0,0 +1,359 @@
+          })
+        }))
+      })
+
+      return function () {
+        subscription.unsubscribe()
+      }
+    },
+    [active, library, account, chainId, setSyncBlock]
+  )
+
+  return { ...syncBlock, syncStatus: syncStatus, setSyncStatus }
+}
+
+const useErc20Token = function (address) {
+  const { active, library } = useWeb3React()
+  return useSWR(active ? address : null, function () {
+    const erc20Service = createErc20(library, address)
+    return Promise.all([
+      erc20Service.symbol(),
+      erc20Service.decimals(),
+      erc20Service.totalSupply()
+    ])
+  })
+}
+
+const Allowance = function ({ address, data }) {
+  const { library } = useWeb3React()
+  const { t } = useTranslation('common')
+
+  const { data: token } = useErc20Token(address)
+
+  if (!token) {
+    return <span className="m-auto"></span>
+  }
+  const [, decimals, totalSupply] = token
+  const allowanceInWei = library.utils.hexToNumberString(data)
+  const value = toFixed(fromUnit(allowanceInWei, decimals), 6)
+  const isUnlimited = Big(totalSupply).times(10).lt(allowanceInWei)
+  return <span className="m-auto">{isUnlimited ? t('unlimited') : value}</span>
+}
+
+const Token = function ({ address }) {
+  const { data: token } = useErc20Token(address)
+  if (!token) {
+    return <span className="m-auto"></span>
+  }
+  const [symbol] = token
+  return <span className="m-auto">{symbol}</span>
+}
+
+const TokenRevokes = function () {
+  const { active } = useWeb3React()
+  const { erc20 } = useContext(PureContext)
+  const { t } = useTranslation('common')
+  const [isRevoking, setIsRevoking] = useState(false)
+
+  const { tokenApprovals, syncStatus, setSyncStatus } = useTokenApprovals()
+
+  const handleRevoke = function (address, spender) {
+    if (!active) {
+      return
+    }
+    setIsRevoking(true)
+    const erc20Service = erc20(address)
+    erc20Service
+      .revoke(spender)
+      .then(() => setIsRevoking(false))
+      .catch(() => setSyncStatus(SyncStatus.Error))
+  }
+
+  return (
+    <Layout walletConnection>
+      {!active && <h3>{t('connect-to-sync')}</h3>}
+      {active && syncStatus === SyncStatus.Syncing && (
+        <h3>{t('syncing-your-approvals')}</h3>
+      )}
+      {active && isRevoking && <h3>{t('revoking-approval')}</h3>}
+      {syncStatus === SyncStatus.Error && <h3>{t('generic-error')}</h3>}
+      {tokenApprovals.length === 0 && syncStatus === SyncStatus.Finished && (
+        <p>{t('no-approvals')}</p>
+      )}
+      {tokenApprovals.length > 0 && (
+        <section className="flex flex-col overflow-x-auto">
+          <h3 className="font-bold text-center text-gray-600">
+            {t('approvals')}
+          </h3>
+          <div className="my-6 grid grid-cols-approval place-content-center gap-y-5 gap-x-12">
+            <span className="m-auto">{t('token')}</span>
+            <span className="m-auto">{t('spender-address')}</span>
+            <span className="m-auto">{t('allowance')}</span>
```

Comment:
The headers should have the a different format (`font-bold text-gray-600`?) to better differentiate those from the table rows.